amazons3/amazons3_connector.bal

```diff
@@ -103,39 +92,33 @@ function AmazonS3Connector::getAllObjects(string bucketName) returns S3Object[]|
     var httpResponse = clientEndpoint->get("/", message = request);
     match httpResponse {
         error err => {
-            amazonS3Error.message = err.message;
-            amazonS3Error.cause = err.cause;
-            return amazonS3Error;
+            return err;
         }
         http:Response response => {
             int statusCode = response.statusCode;
             var amazonResponse = response.getXmlPayload();
             match amazonResponse {
                 error err => {
-                    amazonS3Error.message = err.message;
-                    amazonS3Error.cause = err.cause;
-                    return amazonS3Error;
+                    return err;
                 }
                 xml xmlResponse => {
                     if (statusCode == 200) {
                         return getS3ObjectsList(xmlResponse);
                     }
                     else{
-                        amazonS3Error.message = xmlResponse["Message"].getTextValue();
-                        amazonS3Error.statusCode = statusCode;
-                        return amazonS3Error;
+                        return setResponseError(statusCode, xmlResponse);
                     }
                 }
             }
         }
     }
 }
 
-function AmazonS3Connector::getObject(string bucketName, string objectName) returns S3Object|AmazonS3Error {
+function AmazonS3Connector::getObject(string bucketName, string objectName) returns S3Object|error {
 
     endpoint http:Client clientEndpoint = getClientEndpoint(bucketName);
 
-    AmazonS3Error amazonS3Error = {};
+    error err = {};
```

Comment:
Do we need to create error here, we can move this to line #147 ?

Reply:
Fixed